flink-autoscaler/src/main/java/org/apache/flink/autoscaler/ScalingMetricEvaluator.java

```diff
@@ -296,8 +297,16 @@ protected static void computeProcessingRateThresholds(
             upperUtilization = 1.0;
             lowerUtilization = 0.0;
         } else {
-            upperUtilization = targetUtilization + utilizationBoundary;
-            lowerUtilization = targetUtilization - utilizationBoundary;
+            if (conf.getOptional(UTILIZATION_MAX).isPresent()
+                    || conf.getOptional(UTILIZATION_MIN).isPresent()
+                    || conf.getOptional(TARGET_UTILIZATION_BOUNDARY).isEmpty()) {
+                upperUtilization = conf.get(UTILIZATION_MAX);
+                lowerUtilization = conf.get(UTILIZATION_MIN);
```

Comment:
I'm wondering whether the logic should be the following? 

```suggestion
            if (conf.getOptional(UTILIZATION_MAX).isPresent()
                    && conf.getOptional(UTILIZATION_MIN).isPresent()) {
                upperUtilization = conf.get(UTILIZATION_MAX);
                lowerUtilization = conf.get(UTILIZATION_MIN);
```

Reply:
I think as long as one of UTILIZATION_MAX and UTILIZATION_MIN is set, it means that the user is already using the new config.

At the same time, if the user does not set TARGET_UTILIZATION_BOUNDARY, we still use the new config

Reply:
I think we should not hard-code the defaults in the MIN_MAX options but it should be dynamically computed based on the target.

For example if the users sets target to 0.4 and min to 0.1, then it doesn't make sense to have the default max on 1.. Previously the boundary was 0.4 by default I think we should preserve this

Reply:
I don't know if there is something wrong with my understanding:

Should we keep the boundary and then the  max and min values ​​are derived by `boundary + target` and `target - boundary` by default? Unless the user explicitly sets min max?

If that's the case, I think it's reasonable. 

But in this case, should we change the min max parameter like to `utilization.lower.boundary` and `utilization.upper.boundary` ? 
Then `max = upper.boundary+ target` , `min = target- upper.boundary`

Reply:
As @mxm suggested I think the logic should be enabled if either MIN or MAX is defined. If only one of them is defined then we can compute the default for the other one from the target utilization +- boundary.

I think we should not have default values for the min/max options

Reply:
>As @mxm suggested I think the logic should be enabled if either MIN or MAX is defined

@gyfora Did you mean, if BOTH min and max are defined, the new logic should be enabled? If only one of them is defined, then we compute the other one using the configured target utilization and the boundary. I think that makes sense 👍

Reply:
+1 for not having static defaults for the new options because that would be confusing to users.

Reply:
Thanks @mxm  @gyfora  for review.

> compute the default for the other one from the target utilization +- boundary.

+1 for  this .

I've modified and added compatibility test cases.

PTAL~